### PR TITLE
fix: resolvelocal should always return full ID if image is found (#1563)

### PR DESCRIPTION
**Pull request by iwilltry42**
Ref #1563

### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [x] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [x] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [x] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [x] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

**Comment by cjellick** [comment]
I'm attaching this issue: https://github.com/acorn-io/acorn/issues/1704

The cause of that issue was the original fix to this that was reverted. So, this issue should get sent to QA when this new PR lands.

File: pkg/tags/tags.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ func ResolveLocal(ctx context.Context, c kclient.Client, namespace, image string
 		if IsLocalReference(image) {
 			return "", false, err
 		}
+		return image, false, nil
```
Comment (cjellick) [inline pkg/tags/tags.go:87]: @iwilltry42 is this accurate? you just want This single line change? This does not align with what was in your original change: https://github.com/acorn-io/acorn/pull/1682/files

Comment (thedadams) [reply pkg/tags/tags.go:87]: As I understand it, this is correct.

If the image is not found and not a local reference, then we essentially want to return "this image is not found and that's OK."

By not returning here, we'll go to the `else if !localImage.Remote` branch and essentially return `image, true, nil`.

**Review by thedadams — DISMISSED** [review]

File: pkg/tags/tags.go (hunk 1 above)
Comment (g-linville) [reply pkg/tags/tags.go:87]: But if the first condition is true, won't it just jump to the end of the whole `if else if` chain and return `image, false, nil` anyway?

Comment (thedadams) [reply pkg/tags/tags.go:87]: Oh, geez. You're 100% correct. I had my nesting confused.

Comment (g-linville) [reply pkg/tags/tags.go:87]: Yeah I think that's correct too

Comment (iwilltry42) [reply pkg/tags/tags.go:87]: Huh, that's weird.
It looked different in the diff I first viewed.
It's only this one line but added to my previous change actually.
I just checked it again (on my phone right now) due to your comments and here I don't see the previous changes.
I'll fix it tomorrow morning.

Comment (iwilltry42) [reply pkg/tags/tags.go:87]: I guess it dropped my cherry-pick of the previous change.
It's a Git mystery.

Comment (iwilltry42) [reply pkg/tags/tags.go:87]: Replayed the revert of @cjellick's revert and rebased :heavy_check_mark:

**Review by g-linville — APPROVED** [review]

File: integration/client/apps/apps_test.go [hunk 2]
```diff
@@ -461,3 +462,37 @@ func TestAppRun(t *testing.T) {
...
+		imageID[:8],
+	}
+
+	for _, imageName := range imageNames {
+		_, err := c.AppRun(ctx, imageName, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
```
Comment (njhale) [inline integration/client/apps/apps_test.go:498]: nit: I wonder if we could use sub tests to parallelize these test cases

```suggestion
func TestAppRunImageVariations(t *testing.T) {
	helper.EnsureCRDs(t)
	restConfig := helper.StartAPI(t)

	ctx := helper.GetCTX(t)
	kclient := helper.MustReturn(kclient.Default)
	ns := helper.TempNamespace(t, kclient)

	imageID := client2.NewImage(t, ns.Name)

	c, err := client.New(restConfig, "", ns.Name)
	if err != nil {
		t.Fatal(err)
	}

	if err := c.ImageTag(ctx, imageID, "foo/bar:baz"); err != nil {
		t.Fatal(err)
	}

    for desc, imageName := range map[string]string{
        "ref": "foo/bar:baz",
        "id": imageID,
        "sha256": fmt.Sprintf("sha256:%s", imageID),
        "short": imageID[:8],
    } {
        t.Run(desc, func(t *testing.T) {
             t.Parallel()
             _, err := c.AppRun(ctx, imageName, nil)
             assert.NoError(t, err)
        }
    }
}
```

**Review by njhale — APPROVED** [review]
suggestion on the tests, but totally not pressing

lgtm

File: integration/client/apps/apps_test.go (hunk 2 above)
Comment (njhale) [reply integration/client/apps/apps_test.go:498]: https://github.com/acorn-io/acorn/pull/1731

**Pull request merged**